### PR TITLE
New DropdownMenu component

### PR DESCRIPTION
**Pull request by AtelyPham**
- refactor: the `Button` component
- feat: add new `DropdowmMenu` component

## Summary of changes
- Refactor the `Button` component to accept ref to inner component and give move accessibility
- Add new `DropdownMenu` component
- Add new `MenuItem` component

### Reference issue to close (if applicable)
- Closes https://github.com/webb-tools/webb-dapp/issues/514

-----
### Code Checklist 

- [x] Tested
- [x] Documented

### Screenshots

CleanShot 2022-09-07 at 18 22 03@2x
CleanShot 2022-09-07 at 18 22 16@2x
CleanShot 2022-09-07 at 18 33 00@2x
CleanShot 2022-09-07 at 18 33 10@2x

### CODE REVIEW

File: packages/mixer/src/Mixer.tsx [hunk 1]
```diff
@@ -1,15 +1,75 @@
...
 
 const MixerWrapper = styled.div``;
 type MixerProps = {};
 
 export const Mixer: React.FC<MixerProps> = () => {
+  const dropdownOptions = [
+    { value: 'Github', icon: <GithubFill /> },
+    { value: 'Telegram', icon: <TelegramFill /> },
+    { value: 'Twitter', icon: <TwitterFill /> },
+  ];
```
Comment (AhmedKorim) [inline packages/mixer/src/Mixer.tsx:24]: Could be moved to be constant outside the component  or `useState` in order for it to be instantiated once

**Review by AhmedKorim — APPROVED** [review]
LGMT for a code review , great job @AtelyPham

File: packages/mixer/src/Mixer.tsx (hunk 1 above)
Comment (AhmedKorim) [reply packages/mixer/src/Mixer.tsx:24]: @AtelyPham  please fix the merge conflicts

**Comment by drewstone** [comment]
Need to fix formatting @AtelyPham

**Pull request merged**